packages/client/utils/JiraServerClientManager.ts

```diff
@@ -0,0 +1,49 @@
+import Atmosphere from '../Atmosphere'
+import {MenuMutationProps} from '../hooks/useMutationProps'
+import AddTeamMemberIntegrationAuthMutation from '../mutations/AddTeamMemberIntegrationAuthMutation'
+import getOAuthPopupFeatures from './getOAuthPopupFeatures'
+import CreateOAuth1AuthorizeUrlMutation from '../mutations/CreateOAuth1AuthorizeUrlMutation'
+class JiraServerClientManager {
+  static SCOPES = 'read_api'
+  static openOAuth(
+    atmosphere: Atmosphere,
+    providerId: string,
+    teamId: string,
+    mutationProps: MenuMutationProps
+  ) {
+    const {submitting, onError, onCompleted, submitMutation} = mutationProps
+
+    // Open synchronously because of Safari
+    const popup = window.open(
+      '',
+      'OAuth',
+      getOAuthPopupFeatures({width: 500, height: 750, top: 56})
+    )
+
+    const onUrlCompleted = (result: any) => {
```

Comment:
-1 `const onUrlCompleted = (result: CreateOAuth1AuthorizeUrlMutationResponse) => {`

it's possible that the mutation returns an error & url is null & calling `location.href = null` would result in an error (at least it did for me)